ip_record.js

```diff
@@ -3,36 +3,89 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 // Keep track of events related to just IP addresses
-module.exports = function (BLOCK_INTERVAL_MS, now) {
+module.exports = function (BLOCK_INTERVAL_MS, now, RATE_LIMIT_INTERVAL_MS, MAX_ACCOUNT_STATUS_CHECK) {
 
   now = now || Date.now
 
+  var ACCOUNT_STATUS_ACTION = {
+    requestAccountStatus       : true
+  }
+
   function IpRecord() {}
 
+  function isAccountStatusAction(action) {
+    return ACCOUNT_STATUS_ACTION[action]
+  }
+
   IpRecord.parse = function (object) {
     var rec = new IpRecord()
     object = object || {}
     rec.bk = object.bk // timestamp when the account was blocked
+    rec.as = object.as || []       // timestamps when account status check occurred
+    rec.rl = object.rl       // timestamp when the account was rate-limited
     return rec
   }
 
+  IpRecord.prototype.isOverAccountStatusCheck = function () {
+    return this.as.length > MAX_ACCOUNT_STATUS_CHECK
+  }
+
+  IpRecord.prototype.trimAccountStatus = function (now) {
+    if (this.as.length === 0) { return }
+    // as is naturally ordered from oldest to newest
```

Comment:
nit: quote `'as'` here so it more obviously reads as a name, not a grammatical error :-)